.github/workflows/ci.yaml

```diff
@@ -0,0 +1,94 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  cargo-test:
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Generate Cargo.lock so we can cache from it
+        run: cargo generate-lockfile
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ github.workflow }}-rust-${{ matrix.os }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ github.workflow }}-rust-${{ matrix.os }}-
+      - name: Decrypt test keys
+        run: openssl aes-256-cbc -d -K ${{ secrets.OPENSSL_KEY }} -iv ${{ secrets.OPENSSL_IV }} -in tests/testkeys/iak-dev.pem.enc -out tests/testkeys/iak-dev.pem
+      - name: Cargo test (with one retry on failure)
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 30
+          max_attempts: 2
+          command: IRONCORE_ENV=dev cargo test --features "blocking beta"
```

Comment:
Why do we need to retry? This is a code smell.

Reply:
My thinking here was there we have a fairly large build matrix and a somewhat decent chance that one of the tests will fail due to timeout, internet issues, or similar (and this did occur during my testing). When this failure occurs, it will fail the whole workflow, and there's currently no way to restart individual jobs in a workflow (though many people have asked for that feature, so I have to believe it will come eventually).
So I added a single retry for the test step.
I'm okay with removing it, but if we do I anticipate we'll have some frustration here in the future.
Thoughts?

Reply:
I guess the real code smell here is that our "unit" tests have an external dependency that we're not mocking. Unless somebody wants to somehow mock ironcore-id, or stand up a local one for the tests, your workaround seems like the best option.